### PR TITLE
Feature/nest-server

### PR DESCRIPTION
**Pull request by Nicoss54**
This PR let you to have a nest js server for showing how works cypress with real call

What are the advantages and what are the incovenients

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,43 +1,92 @@
...
+
+### Launch the nest-server project
+
+```bash
+npm run start-nest-server
+```
+
+## Visual studio code user
+
+If you use Vscode, this repo will automatically install the necessary extension if you accepte the proposal.
```
Comment (JonathanGeoffroy) [inline README.md:86]: accepte -> accept

File: README.md [hunk 2]
```diff
@@ -1,43 +1,92 @@
...
+<br/>
+All the necessary task are already created and you can launch this task out of the boxe
```
Comment (JonathanGeoffroy) [inline README.md:88]: All **the** necessary task -> All necessary task**s**

File: exercices/steps/apps/nest-server/src/app/feature/books/books.controller.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+@ApiTags('books')
+@Controller('books')
+export class BooksController {
+  constructor(private readonly booksService: BooksService) {}
+
+  @Get()
+  @ApiOkResponse({ status: 200, description: 'list of books', schema: { type: 'array', items: BOOK_SCHEMA } })
+  getBooks(
+    @Query('page', ParseIntPipe) page: number,
+    @Query('pageSize', ParseIntPipe) pageSize: number,
```
Comment (JonathanGeoffroy) [inline exercices/steps/apps/nest-server/src/app/feature/books/books.controller.ts:17]: Maybe just a details, but I think we should add swagger description (or comment ? ) just to explain that pages are 1-indexed (so first page is '1', not '0')

It's not obvious on the first read, especially if you use the swagger UI
What do you think about it ?

**Review by JonathanGeoffroy — COMMENTED** [review]
Hey ! Thank you so much for this PR ! Great Job !
Just few typos, and I also would like to have a small comment in the swagger if you don't mind, but all the rest is really good  👍 

Thanks again !

File: README.md (hunk 1 above)
Comment (Nicoss54) [reply README.md:86]: done

File: exercices/steps/apps/nest-server/src/app/feature/books/books.controller.ts (hunk 3 above)
Comment (Nicoss54) [reply exercices/steps/apps/nest-server/src/app/feature/books/books.controller.ts:17]: totally agree, il will change that and try to place it on swagger documentation

**Review by JonathanGeoffroy — APPROVED** [review]
Thanks a lot for you work !

**Pull request merged**